src/User/Service/MailService.php

```diff
@@ -83,11 +83,20 @@ public function getType()
      */
     public function run()
     {
-        return $this->mailer
-            ->compose(['html' => $this->view, 'text' => "text/{$this->view}"], $this->params)
-            ->setFrom($this->from)
-            ->setTo($this->to)
-            ->setSubject($this->subject)
-            ->send();
+        try {
+            $result =  $this->mailer
+                ->compose(['html' => $this->view, 'text' => "text/{$this->view}"], $this->params)
+                ->setFrom($this->from)
+                ->setTo($this->to)
+                ->setSubject($this->subject)
+                ->send();
+
+            if (!$result) {
+                Yii::error("Email sending failed to '{$this->to}'.", 'mailer');
+            }
+            return $result;
+        } catch (\Exception $e) {
```

Comment:
Why this catch? Looks like you're trying to capture a configuration error here

Reply:
The issue here was that the error was not clear as the email wasn't being sent correctly. With this we can find what the actual issue is.

In my case the $this->user was not set and in turn $this->to was set to blank

Reply:
If required i will remove the try catch.